### ISSUE TITLE
panic using examples/pcapdump/main.go

### ISSUE THREAD

**Issue by cleesmith** [google/gopacket#175]
```go version go1.6 darwin/amd64```
```
go get -u github.com/google/gopacket
go get -u github.com/google/gopacket/layers
go get -u github.com/google/gopacket/tcpassembly
```
using gopacket/examples/pcapdump/main.go:
```go run pcapdump.go -r test_ethernet.pcap``` ... tried with many pcap files
... output is lots of these:
```%!v(PANIC=reflect.Value.Interface: cannot return value obtained from unexported field or method)```
```.layerString()``` ?
... maybe it's related to this:
https://groups.google.com/forum/#!topic/golang-nuts/iHJcOWmWAag
http://tip.golang.org/doc/go1.6#reflect

other programs work fine ... like: ```httpassembly.go``` and ```bidirectional.go```

**Comment by cleesmith** [comment]
After trying a bunch more pcap files, the following lame fix to ```packet.go``` seems to work:
```
func layerString(...)
   . . .
   case reflect.Struct:
   . . .
     if ftype.Anonymous {
       // anonStr := layerString(f.Interface(), true, writeSpace)
       . . .
       // this seems to only affect output like:
       // 	"Contents=[..20..] Payload=[..1389..]"
       // which appears to be "type BaseLayer struct", so
       // just use an empty string when it's anonymous
       // and that's not a lot of info to lose
       anonStr := layerString("", true, writeSpace)
       . . .
```

**Comment by gconnell** [comment]
Fixed by 445746376b56687f9e29ab50654597a51ff1a489, thanks for the heads up.  You were 100% right, http://tip.golang.org/doc/go1.6#reflect was the cause.
